tests/coalaHTMLTest.py

```diff
@@ -21,8 +24,33 @@ def setUp(self):
 
     def tearDown(self):
         sys.argv = self.old_argv
+        """
+        Before launching the server, we `chdir` to angular app i.e
+        `coalahtmlapp` in coala_html.main. So we need to reset this so
+        that other tests can run fine.
+        """
+        os.chdir(os.path.dirname(os.path.dirname(__file__)))
+        shutil.rmtree('coalahtmlapp', ignore_errors=True)
 
-    def test_output_file(self):
+    @patch.object(socketserver.TCPServer, 'serve_forever', autospec=True)
+    @patch.object(socketserver.TCPServer, 'server_close', autospec=True)
+    @patch('coalahtml.coala_html.call')
+    @patch('webbrowser.open')
+    def test_output_file(self, mock_browser,
+                         mock_subprocess_call,
+                         mock_server_close,
+                         mock_serve_forever):
+        """
+        The approach is to call ``coala_html.main`` twice - first with
+        nolaunch and then with noupdate args. With only nolaunch being
+        passed, coala_html.main would generate the json configs and next
+        time we needn't update and just test for old json files running 
+        ``coala_html.main`` with noupdate. In the process we mock the
+        expensive http calls and instantiating the server by mocking the 
+        corresponding method calls.
```

Comment:
Make it shorter, like:
```
Test that when JSON configs are already generated, coala-html with 'noupdate' option will run successfully and not update existing configuration.
```
Something like that. And change the name of the test, so it'll be closer to what behavior you're actually testing.